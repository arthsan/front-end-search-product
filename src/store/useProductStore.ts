import create from 'zustand'

export type Product = {
  _meta: Meta
  id?: string
  name: string
  type: string
}

export type Meta = {
  score: number
  visitsClickCount: string
}

type State = {
  products: Product[]
  addProduct: (product: Product) => void
}

export const useProductStore = create<State>((set) => ({
  products: [],

  addProduct: async (product: Product) => {
    await set((state) => ({ products: [...state.products, product] }))
  }
}))
