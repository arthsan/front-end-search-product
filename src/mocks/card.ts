import { Product } from 'store/useProductStore'

export const cardMock: Product = {
  id: 'newID',
  name: 'Camiseta',
  type: 'organic',
  _meta: {
    score: 10,
    visitsClickCount: '100'
  }
}
