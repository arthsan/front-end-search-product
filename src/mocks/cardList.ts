import { Product } from 'store/useProductStore'

export const cardList: Product[] = [
  {
    name: 'Kit Com 5 Camisetas Básicas Esportivas Masculina Preta, Branca E Cinza',
    id: '56664056',
    _meta: {
      score: 345.77496,
      visitsClickCount: '779'
    },
    type: 'organic'
  },
  {
    name: 'Camiseta masculina jordam 100% algodão',
    id: '4116684846',
    _meta: {
      score: 212.25824,
      visitsClickCount: '246'
    },
    type: 'organic'
  },
  {
    name: 'Kit Com 5 Camisetas Básicas Part. B Masculina Algodão T-shirt Tee',
    id: '80502107',
    _meta: {
      score: 199.59839,
      visitsClickCount: '445'
    },
    type: 'organic'
  },
  {
    name: 'Camiseta Estampada Plus Size Masculina 1000075054 Malwee Wee',
    id: '2533730399',
    _meta: {
      score: 123.03333,
      visitsClickCount: '158'
    },
    type: 'organic'
  },
  {
    name: 'Kit 5 Camisetas Básicas Masculina 100% Poliéster Cores Sortidas',
    id: '3374278226',
    _meta: {
      score: 110.10536,
      visitsClickCount: '131'
    },
    type: 'organic'
  },
  {
    name: 'Camiseta Masculina Personalizada Stranger Things Série',
    id: '2418950425',
    _meta: {
      score: 105.30116,
      visitsClickCount: '183'
    },
    type: 'organic'
  }
]
