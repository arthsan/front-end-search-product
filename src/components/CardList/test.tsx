import { screen, render } from '@testing-library/react'
import { useProductStore } from 'store/useProductStore'

import { CardList } from '.'

import { cardMock } from '../../mocks/card'
import { cardList } from '../../mocks/cardList'

const originalState = useProductStore.getState()
beforeEach(() => {
  useProductStore.setState(originalState)
  cardList.map((card) => {
    originalState.addProduct(card)
  })
})

describe('<CardList />', () => {
  it('should render first product title ', () => {
    const { container } = render(<CardList />)

    const listElements = screen.getAllByRole('product')
    expect(listElements[0].firstChild).toHaveTextContent(cardList[0].name)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a list of 6 products', () => {
    const { container } = render(<CardList />)

    const listElements = screen.getAllByRole('product')
    expect(listElements).toHaveLength(6)
    expect(container.firstChild).toMatchSnapshot()
  })
})
