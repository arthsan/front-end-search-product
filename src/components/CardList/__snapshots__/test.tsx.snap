// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> should render a list of products 1`] = `
<div
  class="css-miwkxx"
>
  <div
    class="css-0"
    role="product"
  >
    <div
      class="css-njtlda"
    >
      <h2
        class="chakra-heading css-1s3enhb"
        role="title"
      >
        Kit Com 5 Camisetas Básicas Esportivas Masculina Preta, Branca E Cinza
      </h2>
      <p
        class="chakra-text css-in3yi3"
        role="description"
      >
        organic
      </p>
      <p
        class="chakra-text css-20qhnm"
      >
        Store Ranking 
        345.77496
      </p>
      <p
        class="chakra-text css-0"
        role="views"
      >
        Public views 
        779
      </p>
    </div>
  </div>
  <div
    class="css-0"
    role="product"
  >
    <div
      class="css-njtlda"
    >
      <h2
        class="chakra-heading css-1s3enhb"
        role="title"
      >
        Camiseta masculina jordam 100% algodão
      </h2>
      <p
        class="chakra-text css-in3yi3"
        role="description"
      >
        organic
      </p>
      <p
        class="chakra-text css-20qhnm"
      >
        Store Ranking 
        212.25824
      </p>
      <p
        class="chakra-text css-0"
        role="views"
      >
        Public views 
        246
      </p>
    </div>
  </div>
  <div
    class="css-0"
    role="product"
  >
    <div
      class="css-njtlda"
    >
      <h2
        class="chakra-heading css-1s3enhb"
        role="title"
      >
        Kit Com 5 Camisetas Básicas Part. B Masculina Algodão T-shirt Tee
      </h2>
      <p
        class="chakra-text css-in3yi3"
        role="description"
      >
        organic
      </p>
      <p
        class="chakra-text css-20qhnm"
      >
        Store Ranking 
        199.59839
      </p>
      <p
        class="chakra-text css-0"
        role="views"
      >
        Public views 
        445
      </p>
    </div>
  </div>
  <div
    class="css-0"
    role="product"
  >
    <div
      class="css-njtlda"
    >
      <h2
        class="chakra-heading css-1s3enhb"
        role="title"
      >
        Camiseta Estampada Plus Size Masculina 1000075054 Malwee Wee
      </h2>
      <p
        class="chakra-text css-in3yi3"
        role="description"
      >
        organic
      </p>
      <p
        class="chakra-text css-20qhnm"
      >
        Store Ranking 
        123.03333
      </p>
      <p
        class="chakra-text css-0"
        role="views"
      >
        Public views 
        158
      </p>
    </div>
  </div>
  <div
    class="css-0"
    role="product"
  >
    <div
      class="css-njtlda"
    >
      <h2
        class="chakra-heading css-1s3enhb"
        role="title"
      >
        Kit 5 Camisetas Básicas Masculina 100% Poliéster Cores Sortidas
      </h2>
      <p
        class="chakra-text css-in3yi3"
        role="description"
      >
        organic
      </p>
      <p
        class="chakra-text css-20qhnm"
      >
        Store Ranking 
        110.10536
      </p>
      <p
        class="chakra-text css-0"
        role="views"
      >
        Public views 
        131
      </p>
    </div>
  </div>
  <div
    class="css-0"
    role="product"
  >
    <div
      class="css-njtlda"
    >
      <h2
        class="chakra-heading css-1s3enhb"
        role="title"
      >
        Camiseta Masculina Personalizada Stranger Things Série
      </h2>
      <p
        class="chakra-text css-in3yi3"
        role="description"
      >
        organic
      </p>
      <p
        class="chakra-text css-20qhnm"
      >
        Store Ranking 
        105.30116
      </p>
      <p
        class="chakra-text css-0"
        role="views"
      >
        Public views 
        183
      </p>
    </div>
  </div>
</div>
`;

exports[`<CardList /> should render first product title  1`] = `
<div
  class="css-miwkxx"
>
  <div
    class="css-0"
    role="product"
  >
    <div
      class="css-njtlda"
    >
      <h2
        class="chakra-heading css-1s3enhb"
        role="title"
      >
        Kit Com 5 Camisetas Básicas Esportivas Masculina Preta, Branca E Cinza
      </h2>
      <p
        class="chakra-text css-in3yi3"
        role="description"
      >
        organic
      </p>
      <p
        class="chakra-text css-20qhnm"
      >
        Store Ranking 
        345.77496
      </p>
      <p
        class="chakra-text css-0"
        role="views"
      >
        Public views 
        779
      </p>
    </div>
  </div>
  <div
    class="css-0"
    role="product"
  >
    <div
      class="css-njtlda"
    >
      <h2
        class="chakra-heading css-1s3enhb"
        role="title"
      >
        Camiseta masculina jordam 100% algodão
      </h2>
      <p
        class="chakra-text css-in3yi3"
        role="description"
      >
        organic
      </p>
      <p
        class="chakra-text css-20qhnm"
      >
        Store Ranking 
        212.25824
      </p>
      <p
        class="chakra-text css-0"
        role="views"
      >
        Public views 
        246
      </p>
    </div>
  </div>
  <div
    class="css-0"
    role="product"
  >
    <div
      class="css-njtlda"
    >
      <h2
        class="chakra-heading css-1s3enhb"
        role="title"
      >
        Kit Com 5 Camisetas Básicas Part. B Masculina Algodão T-shirt Tee
      </h2>
      <p
        class="chakra-text css-in3yi3"
        role="description"
      >
        organic
      </p>
      <p
        class="chakra-text css-20qhnm"
      >
        Store Ranking 
        199.59839
      </p>
      <p
        class="chakra-text css-0"
        role="views"
      >
        Public views 
        445
      </p>
    </div>
  </div>
  <div
    class="css-0"
    role="product"
  >
    <div
      class="css-njtlda"
    >
      <h2
        class="chakra-heading css-1s3enhb"
        role="title"
      >
        Camiseta Estampada Plus Size Masculina 1000075054 Malwee Wee
      </h2>
      <p
        class="chakra-text css-in3yi3"
        role="description"
      >
        organic
      </p>
      <p
        class="chakra-text css-20qhnm"
      >
        Store Ranking 
        123.03333
      </p>
      <p
        class="chakra-text css-0"
        role="views"
      >
        Public views 
        158
      </p>
    </div>
  </div>
  <div
    class="css-0"
    role="product"
  >
    <div
      class="css-njtlda"
    >
      <h2
        class="chakra-heading css-1s3enhb"
        role="title"
      >
        Kit 5 Camisetas Básicas Masculina 100% Poliéster Cores Sortidas
      </h2>
      <p
        class="chakra-text css-in3yi3"
        role="description"
      >
        organic
      </p>
      <p
        class="chakra-text css-20qhnm"
      >
        Store Ranking 
        110.10536
      </p>
      <p
        class="chakra-text css-0"
        role="views"
      >
        Public views 
        131
      </p>
    </div>
  </div>
  <div
    class="css-0"
    role="product"
  >
    <div
      class="css-njtlda"
    >
      <h2
        class="chakra-heading css-1s3enhb"
        role="title"
      >
        Camiseta Masculina Personalizada Stranger Things Série
      </h2>
      <p
        class="chakra-text css-in3yi3"
        role="description"
      >
        organic
      </p>
      <p
        class="chakra-text css-20qhnm"
      >
        Store Ranking 
        105.30116
      </p>
      <p
        class="chakra-text css-0"
        role="views"
      >
        Public views 
        183
      </p>
    </div>
  </div>
</div>
`;
